### ISSUE TITLE
Add shuffle to SC data iterator

### ISSUE THREAD

**Issue by fareeha** [apple/turicreate#2633]
The `_NumPyDataIterator` needs shuffle implemented.

**Comment by TobyRoseman** [comment]
@fareeha - please give more information here. Is this shuffling the order of the deep feature, prior to training the custom neural network? Is this something we had prior to migrating to TensorFlow?

**Comment by fareeha** [comment]
`_NumPyDataIterator` is a data iterator used to create batches for TF to perform train. We have yet to add the option to `shuffle`. This issue is created to track that.

MXNet data iterator had the option to `shuffle`.

**Comment by TobyRoseman** [comment]
I think it might make more sense to implement a `shuffle` method for SFrame (i.e. issue #942) and then just use that for this method.

**Comment by fareeha** [comment]
> I think it might make more sense to implement a `shuffle` method for SFrame (i.e. issue #942) and then just use that for this method.

I agree! Shuffle on SFrame would be the right place to put this effort!

**Comment by TobyRoseman** [comment]
SFrame now has a [shuffle method](https://apple.github.io/turicreate/docs/api/generated/turicreate.SFrame.shuffle.html#turicreate.SFrame.shuffle). So this should be fairly straightforward.

However let's make sure to benchmark this change, making sure the runtime increase is worth the performance improvements.
